internal/corazawaf/transaction_test.go

```diff
@@ -1477,24 +1568,71 @@ func TestForceRequestBodyOverride(t *testing.T) {
 	tx.variables.RequestBodyProcessor().(*collections.Single).Set("JSON")
 	tx.ProcessRequestHeaders()
 	if _, _, err := tx.WriteRequestBody([]byte("foo=bar&baz=qux")); err != nil {
-		t.Errorf("Failed to write request body: %v", err)
+		t.Fatalf("Failed to write request body: %v", err)
 	}
 	if _, err := tx.ProcessRequestBody(); err != nil {
-		t.Errorf("Failed to process request body: %v", err)
+		t.Fatalf("Failed to process request body: %v", err)
 	}
 	if tx.variables.RequestBodyProcessor().Get() != "JSON" {
-		t.Errorf("Failed to force request body variable")
+		t.Fatalf("Failed to force request body variable")
 	}
 	tx = waf.NewTransaction()
 	tx.ForceRequestBodyVariable = true
 	tx.ProcessRequestHeaders()
 	if _, _, err := tx.WriteRequestBody([]byte("foo=bar&baz=qux")); err != nil {
-		t.Errorf("Failed to write request body: %v", err)
+		t.Fatalf("Failed to write request body: %v", err)
 	}
 	if _, err := tx.ProcessRequestBody(); err != nil {
-		t.Errorf("Failed to process request body: %v", err)
+		t.Fatalf("Failed to process request body: %v", err)
 	}
 	if tx.variables.RequestBodyProcessor().Get() != "URLENCODED" {
-		t.Errorf("Failed to force request body variable, got RBP: %q", tx.variables.RequestBodyProcessor().Get())
+		t.Fatalf("Failed to force request body variable, got RBP: %q", tx.variables.RequestBodyProcessor().Get())
+	}
+
+	if err := tx.Close(); err != nil {
+		t.Fatalf("failed to close transaction: %s", err.Error())
+	}
+}
+
+func TestCloseFails(t *testing.T) {
+	waf := NewWAF()
+	tx := waf.NewTransaction()
+	tx.Variables().FilesTmpContent().Set("foo", []string{"unexisting"})
```

Comment:
Insterestingly this test is failing because the collection is always empty. Any clue @anuraaga @jptosso ?